src/transports/PostMessageIframeTransport.ts

```diff
@@ -46,7 +47,7 @@ class PostMessageIframeTransport extends Transport {
     const prom = this.transportRequestManager.addRequest(data, null);
     const notifications = getNotifications(data);
     if (this.frame) {
-      this.frame.postMessage((data as IJSONRPCData).request, "*");
+      this.frame.postMessage((data as IJSONRPCData).request, this.uri);
```

Comment:
```suggestion
      this.frame.postMessage((data as IJSONRPCData).request, this.origin);
```

Reply:
the origin can easily be derived as new URL(this.uri).origin
We don't need another parameter from ctor